README.md
# Date Constants
[![Build Status](https://travis-ci.org/repat/date-constants.svg?branch=develop)](https://travis-ci.org/repat/date-constants)
[![Packagist](https://img.shields.io/packagist/v/repat/date-constants.svg)](https://packagist.org/packages/repat/date-constants)

The **Date Constants** package defines a number of useful PHP constants within your application, making it easier to express measures of dates.

These constants may seem familiar to Laravel developers, as they're absolutely [inspired by Nesbot\Carbons use of date constants](https://carbon.nesbot.com/docs/#api-constants). Also, shoutout to the [`stevegrunwell/time-constants`](https://github.com/stevegrunwell/time-constants) package for the idea.

## Installation
Date Constants is intended to be installed via Composer:

```sh
$ composer require repat/date-constants
```

The package has been configured to automatically expose the `constants.php` file via the Composer-generated autoloader, so you'll be ready to use the constants as soon as the package has been installed!

## Defined constants
This is a list of all constants defined by this package, along with their values. Each constant is wrapped in a `if (! defined(...))` conditional, ensuring these constants can easily be redefined if necessary and won't conflict with existing constants.

### From Carbon
* `YEARS_PER_CENTURY` (100)
* `YEARS_PER_DECADE` (10)
* `MONTHS_PER_YEAR` (12)
* `WEEKS_PER_YEAR` (52)
* `DAYS_PER_WEEK` (7)
* `HOURS_PER_DAY` (24)

#### Days of the week
* `SUNDAY` (0)
* `MONDAY` (1)
* `TUESDAY` (2)
* `WEDNESDAY` (3)
* `THURSDAY` (4)
* `FRIDAY` (5)
* `SATURDAY` (6)

### Misc
* `DAYS_PER_YEAR` (365)
* `DAYS_PER_LEAP_YEAR` (366)
* `DAYS_PER_FORTNIGHT` (14)

### Months
* `JANUARY` (1)
* `FEBRUARY` (2)
* `MARCH` (3)
* `APRIL` (4)
* `MAY` (5)
* `JUNE` (6)
* `JULY` (7)
* `AUGUST` (8)
* `SEPTEMBER` (9)
* `OCTOBER` (19)
* `NOVEMBER` (11)
* `DECEMBER` (12)

#### Amount of days
* `DAYS_JAN` (31)
* `DAYS_FEB` (28)
* `DAYS_FEB_LEAP` (29)
* `DAYS_MAR` (31)
* `DAYS_APR` (30)
* `DAYS_JUN` (31)
* `DAYS_JUL` (30)
* `DAYS_AUG` (31)
* `DAYS_SEP` (30)
* `DAYS_OCT` (31)
* `DAYS_NOV` (30)
* `DAYS_DEC` (31)


## TODO
* Finish Tests


## Version
* Version 2.0.1

## Contact
#### repat
* Homepage: https://repat.de
* e-mail: repat@repat.de
* Twitter: [@repat123](https://twitter.com/repat123 "repat123 on twitter")

[![Flattr this git repo](http://api.flattr.com/button/flattr-badge-large.png)](https://flattr.com/submit/auto?user_id=repat&url=https://github.com/repat/date-constants&title=date-constants&language=&tags=github&category=software)


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Version 2.0.0] - 2019-04-18

* Forked `stevegrunwell/time-constants` and added first date constants

## [Version 1.1.0] - 2019-01-01

* Addition of both `ONE_SECOND` and `ONE_MINUTE`, which are both equal to `1` ([#2]).
* Conform to the PSR-2 coding standards ([#3]).
* Adjustments to the PHPUnit and Travis CI configuration files.

## [Version 1.0.0] - 2018-03-21

Initial public release of the library, with the following constants:

* `MINUTE_IN_SECONDS` (60 seconds)
* `HOUR_IN_SECONDS` (3600 seconds)
* `DAY_IN_SECONDS` (86,400 seconds)
* `WEEK_IN_SECONDS` (604,800 seconds)
* `MONTH_IN_SECONDS` (2,592,000 seconds)
* `YEAR_IN_SECONDS` (31,536,000 seconds)
* `HOUR_IN_MINUTES` (60 minutes)
* `DAY_IN_MINUTES` (1,440 minutes)
* `WEEK_IN_MINUTES` (10,080 minutes)
* `MONTH_IN_MINUTES` (43,200 minutes)
* `YEAR_IN_MINUTES` (525,600 minutes)


[Unreleased]: https://github.com/stevegrunwell/time-constants/compare/master...develop
[Version 1.1.0]: https://github.com/stevegrunwell/time-constants/releases/tag/v1.1.0
[Version 1.0.0]: https://github.com/stevegrunwell/time-constants/releases/tag/v1.0.0
[#2]: https://github.com/stevegrunwell/time-constants/pull/2
[#3]: https://github.com/stevegrunwell/time-constants/pull/3


tests/ConstantsTest.php
<?php
/**
 * Tests the definition of time constants.
 *
 * @package SteveGrunwell\TimeConstants
 */

use PHPUnit\Framework\TestCase;

class ConstantsTest extends TestCase
{
    /**
     * Ensure that each constant is defined and numeric.
     *
     * @dataProvider constantsProvider()
     */
    public function testConstantsAreDefined($constant)
    {
        $this->assertTrue(defined($constant), 'Expected the constant to be defined.');
        $this->assertInternalType('integer', constant($constant), 'Expected an integer value.');
    }

    /**
     * Provides a list of all constants defined by this package.
     */
    public function constantsProvider()
    {
        return [
            'January'       => ['JANUARY'],
            'February'       => ['FEBRUARY'],
            'March'       => ['MARCH'],
            'April'       => ['APRIL'],
            'May'       => ['MAY'],
            'June'       => ['JUNE'],
            'July'       => ['JULY'],
            'August'       => ['AUGUST'],
            'September'       => ['SEPTEMBER'],
            'October'       => ['OCTOBER'],
            'November'       => ['NOVEMBER'],
            'December'       => ['DECEMBER'],
        ];
    }
}


constants.php
<?php
/**
 * Defines useful date-based constants.
 *
 * @package Repat\DateConstants
 */

/*
|--------------------------------------------------------------------------
| From Nesbot\Carbon
|--------------------------------------------------------------------------
*/

if (! defined('YEARS_PER_CENTURY')) {
    define('YEARS_PER_CENTURY', 100);
}

if (! defined('YEARS_PER_DECADE')) {
    define('YEARS_PER_DECADE', 10);
}

if (! defined('MONTHS_PER_YEAR')) {
    define('MONTHS_PER_YEAR', 12);
}

if (! defined('WEEKS_PER_YEAR')) {
    define('WEEKS_PER_YEAR', 52);
}

if (! defined('DAYS_PER_WEEK')) {
    define('DAYS_PER_WEEK', 7);
}

if (! defined('HOURS_PER_DAY')) {
    define('HOURS_PER_DAY', 24);
}

/*
|--------------------------------------------------------------------------
| Days of the week from Carbon
|--------------------------------------------------------------------------
*/

if (! defined('SUNDAY')) {
    define('SUNDAY', 0);
}

if (! defined('MONDAY')) {
    define('MONDAY', 1);
}

if (! defined('TUESDAY')) {
    define('TUESDAY', 2);
}

if (! defined('WEDNESDAY')) {
    define('WEDNESDAY', 3);
}

if (! defined('THURSDAY')) {
    define('THURSDAY', 4);
}

if (! defined('FRIDAY')) {
    define('FRIDAY', 5);
}

if (! defined('SATURDAY')) {
    define('SATURDAY', 6);
}

/*
|--------------------------------------------------------------------------
| Misc
|--------------------------------------------------------------------------
*/

if (! defined('DAYS_PER_YEAR')) {
    define('DAYS_PER_YEAR', 365);
}

if (! defined('DAYS_PER_LEAP_YEAR')) {
    define('DAYS_PER_LEAP_YEAR', 366);
}

if (! defined('DAYS_PER_FORTNIGHT')) {
    define('DAYS_PER_FORTNIGHT', 14);
}

/*
|--------------------------------------------------------------------------
| Months
|--------------------------------------------------------------------------
*/

if (! defined('JANUARY')) {
    define('JANUARY', 1);
}

if (! defined('FEBRUARY')) {
    define('FEBRUARY', 2);
}

if (! defined('MARCH')) {
    define('MARCH', 3);
}

if (! defined('APRIL')) {
    define('APRIL', 4);
}

if (! defined('MAY')) {
    define('MAY', 5);
}

if (! defined('JUNE')) {
    define('JUNE', 6);
}

if (! defined('JULY')) {
    define('JULY', 7);
}

if (! defined('AUGUST')) {
    define('AUGUST', 8);
}

if (! defined('SEPTEMBER')) {
    define('SEPTEMBER', 9);
}

if (! defined('OCTOBER')) {
    define('OCTOBER', 10);
}

if (! defined('NOVEMBER')) {
    define('NOVEMBER', 11);
}

if (! defined('DECEMBER')) {
    define('DECEMBER', 12);
}

/*
|--------------------------------------------------------------------------
| Days of Months
|--------------------------------------------------------------------------
*/

if (! defined('DAYS_JAN')) {
    define('DAYS_JAN', 31);
}

if (! defined('DAYS_FEB')) {
    define('DAYS_FEB', 28);
}

if (! defined('DAYS_FEB_LEAP')) {
    define('DAYS_FEB_LEAP', 29);
}

if (! defined('DAYS_MAR')) {
    define('DAYS_MAR', 31);
}

if (! defined('DAYS_APR')) {
    define('DAYS_APR', 30);
}

if (! defined('DAYS_JUN')) {
    define('DAYS_JUN', 31);
}

if (! defined('DAYS_JUL')) {
    define('DAYS_JUL', 30);
}

if (! defined('DAYS_AUG')) {
    define('DAYS_AUG', 31);
}

if (! defined('DAYS_SEP')) {
    define('DAYS_SEP', 30);
}

if (! defined('DAYS_OCT')) {
    define('DAYS_OCT', 31);
}

if (! defined('DAYS_NOV')) {
    define('DAYS_NOV', 30);
}

if (! defined('DAYS_DEC')) {
    define('DAYS_DEC', 31);
}
